dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/am/KubernetesApplicationManager.java

```diff
@@ -180,16 +182,25 @@ private TaskExecutionStatus getApplicationStatus(KubernetesApplicationManagerCon
      */
     public LogWatch getPodLogWatcher(KubernetesApplicationManagerContext kubernetesApplicationManagerContext) {
         KubernetesClient client = getClient(kubernetesApplicationManagerContext);
-        FilterWatchListDeletable<Pod, PodList, PodResource> watchList =
-                getDriverPod(kubernetesApplicationManagerContext);
-        List<Pod> driverPod = watchList.list().getItems();
-        if (CollectionUtils.isEmpty(driverPod)) {
-            return null;
+        boolean podIsReady = false;
+        Pod pod = null;
+        while (!podIsReady) {
+            FilterWatchListDeletable<Pod, PodList, PodResource> watchList =
+                    getListenPod(kubernetesApplicationManagerContext);
+            List<Pod> podList = watchList.list().getItems();
+            if (CollectionUtils.isEmpty(podList)) {
+                return null;
```

Comment:
Should not return directly, perhaps the k8s controller manager is currently under high system load, and the scheduling is a little slow.

We should wait and retry, but of course we should set a reasonable number of retries.

Reply:
Really appreciate, I'll make some changes later today :D